src/vs/workbench/api/node/extHostLanguageFeatures.ts

```diff
@@ -214,6 +214,33 @@ class ReferenceAdapter {
 	}
 }
 
+class ExternalReferenceAdapter {
+
+	constructor(private _provider: vscode.ExternalReferenceProvider) {
+	}
+
+	provideExternalReferences(query: modes.ISymbolDescriptor, hints: { [hint: string]: any }, progress: (references: modes.IReferenceInformation[]) => void): TPromise<modes.IReferenceInformation[]> {
+		const convertReferences = (references: vscode.ReferenceInformation[]) => {
```

Comment:
kind of pointless. the type system "knows" `references` is an array so it concludes that this method can never return undefined. Even with the explicit `| undefined` the type of convertReferences is always just `IReferenceInformation[]`